api/resource_groups.go

```diff
@@ -0,0 +1,236 @@
+	if err != nil {
+		return ResourceGroupResponse{}, err
+	}
+
+	return setResourceGroupResponse(rawResponse), nil
+}
+
+func setResourceGroupResponse(response ResourceGroupCreateResponse) ResourceGroupResponse {
+	return ResourceGroupResponse{
+		Data: ResourceGroupData{
+			Guid:         response.Data.Guid,
+			IsDefault:    strconv.Itoa(response.Data.IsDefault),
+			ResourceGuid: response.Data.ResourceGuid,
+			Name:         response.Data.Name,
+			Type:         response.Data.Type,
+			Enabled:      response.Data.Enabled,
+			Props:        response.Data.Props,
+		},
+	}
+}
+
+// Delete deletes a Resource Group that matches the provided resource guid
+func (svc *ResourceGroupsService) Delete(guid string) error {
+	if guid == "" {
+		return errors.New("specify a resourceGuid")
+	}
+
+	return svc.client.RequestDecoder(
+		"DELETE",
+		fmt.Sprintf(apiV2ResourceGroupsFromGUID, guid),
+		nil,
+		nil,
+	)
+}
+
+// Get returns a raw response of the Resource Group with the matching resource guid.
+//
+// To return a more specific Go struct of a Resource Group, use the proper
+// method such as GetContainerResourceGroup() where the function name is composed by:
+//
+//  Get<Type>(guid)
+//
+//    Where <Type> is the Resource Group type.
+func (svc *ResourceGroupsService) Get(guid string, response interface{}) error {
+	return svc.get(guid, &response)
+}
+
+func (svc *ResourceGroupsService) create(data interface{}, response interface{}) error {
+	return svc.client.RequestEncoderDecoder("POST", apiV2ResourceGroups, data, response)
+}
+
+func (svc *ResourceGroupsService) get(guid string, response interface{}) error {
+	if guid == "" {
+		return errors.New("specify an resourceGuid")
+	}
+	apiPath := fmt.Sprintf(apiV2ResourceGroupsFromGUID, guid)
+	return svc.client.RequestDecoder("GET", apiPath, nil, response)
+}
+
+func (svc *ResourceGroupsService) update(guid string, data interface{}, response interface{}) error {
+	if guid == "" {
+		return errors.New("specify an intgGuid")
+	}
+	apiPath := fmt.Sprintf(apiV2ResourceGroupsFromGUID, guid)
+	return svc.client.RequestEncoderDecoder("PATCH", apiPath, data, response)
+}
+
+func (group ResourceGroupData) ResourceGroupType() ResourceGroupType {
+	t, _ := FindResourceGroupType(group.Type)
+	return t
+}
+
+func (group ResourceGroupData) ID() string {
+	return group.ResourceGuid
+}
+
+type ResourceGroupResponse struct {
+	Data ResourceGroupData `json:"data"`
+}
+
+type ResourceGroupsResponse struct {
+	Data []ResourceGroupData `json:"data"`
+}
+
+type ResourceGroupData struct {
+	Guid         string      `json:"guid,omitempty"`
+	IsDefault    string      `json:"isDefault,omitempty"`
+	ResourceGuid string      `json:"resourceGuid,omitempty"`
+	Name         string      `json:"resourceName"`
+	Type         string      `json:"resourceType"`
+	Enabled      int         `json:"enabled,omitempty"`
+	Props        interface{} `json:"props"`
+}
+
+// RAIN-21510 workaround
+type ResourceGroupCreateResponse struct {
```

Comment:
Lets make this struct private:
```suggestion
type resourceGroupCreateResponse struct {
```